master/buildbot/test/unit/test_scripts_cleanupdb.py

```diff
@@ -43,6 +43,19 @@ def mkconfig(**kwargs):
     return config
 
 
+def patch_environ(case, key, value):
+    """
+    Add an environment variable for the duration of a test.
+    """
+    old_environ = os.environ.copy()
+
+    def cleanup():
+        os.environ.clear()
+        os.environ.update(old_environ)
+    os.environ[key] = value
+    case.addCleanup(cleanup)
+
+
 class TestCleanupDb(misc.StdoutAssertionsMixin, dirs.DirsMixin,
                     db.RealDatabaseMixin, unittest.TestCase):
```

Comment:
I prefer to avoid inheritance if I can help it. Making this into a mixin doesn't provide any benefit over having a free function, so I just went with a free function.